prism/src/main/java/org/apache/falcon/service/FeedSLAAlertService.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.falcon.resource.SchedulableEntityInstance;
+import org.apache.falcon.util.ReflectionUtils;
+import org.apache.falcon.util.StartupProperties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+  * Service to know which all feeds have missed SLA.
+  */
+public final class FeedSLAAlertService implements FalconService, EntitySLAListener {
+
+    private static final String NAME = "FeedSLAAlertService";
+
+    private static final Logger LOG = LoggerFactory.getLogger(FeedSLAAlertService.class);
+
+    private MonitoringJdbcStateStore store = new MonitoringJdbcStateStore();
+
+    private Set<EntitySLAListener> listeners = new LinkedHashSet<EntitySLAListener>();
+
+    private static final FeedSLAAlertService SERVICE = new FeedSLAAlertService();
+
+    public static FeedSLAAlertService get() {
+        return SERVICE;
+    }
+
+    private FeedSLAAlertService(){}
+
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public void init() throws FalconException {
+        String listenerClassNames = StartupProperties.get().
+                getProperty("feedAlert.listeners");
+        for (String listenerClassName : listenerClassNames.split(",")) {
+            listenerClassName = listenerClassName.trim();
+            if (listenerClassName.isEmpty()) {
+                continue;
+            }
+            EntitySLAListener listener = ReflectionUtils.getInstanceByClassName(listenerClassName);
+            registerListener(listener);
+        }
+
+        String freq = StartupProperties.get().getProperty("feed.sla.statusCheck.frequency.seconds", "600");
+        int statusCheckFrequencySeconds = Integer.parseInt(freq);
+
+        ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1);
+        executor.scheduleWithFixedDelay(new Monitor(), 0, statusCheckFrequencySeconds + 10, TimeUnit.SECONDS);
+    }
+
+    public void registerListener(EntitySLAListener listener) {
+        listeners.add(listener);
+    }
+
+    @Override
+    public void destroy() throws FalconException {
+
+    }
+
+
+    private class Monitor implements Runnable {
+
+        @Override
+        public void run() {
+            processSLACandidates();
+        }
+    }
+
+    void processSLACandidates(){
+        Multimap<Pair, Date> feedInstancesToBeMonitored = ArrayListMultimap.create();
+//Get all feeds instances to be monitored
+        List<PendingInstanceBean> pendingInstanceBeanList = store.getAllInstances();
+        if (pendingInstanceBeanList.isEmpty()){
+            return;
+        }
+
+        LOG.debug("In processSLACandidates :" + pendingInstanceBeanList.size());
+        try{
+            for (PendingInstanceBean pendingInstanceBean : pendingInstanceBeanList) {
+                Pair pair = new Pair(pendingInstanceBean.getFeedName(), pendingInstanceBean.getClusterName());
+                feedInstancesToBeMonitored.put(pair, pendingInstanceBean.getNominalTime());
```

Comment:
Do not see the need to create this Map and then iterate over it. Can't we just iterate over pendingInstanceBeanList instead?